tests/vxlan/vnet_utils.py

```diff
@@ -0,0 +1,122 @@
+    @param **template_kwargs: Any keyword arguments to be passed to j2 during rendering
+    """
+    # Combine all dictionaries given in template_args
+    template_args_iter = iter(template_args)
+    try:
+        combined_template_args = next(template_args_iter).copy()
+        for arg in template_args_iter:
+            combined_template_args.update(arg)
+    except StopIteration:
+        combined_template_args = {}
+
+    rendered = Template(open(path.join("templates",template_name)).read()) \
+                        .render(combined_template_args, **template_kwargs)
+
+    host.copy(content=rendered, dest=dest_file)
+
+def generate_dut_config_files(duthost, mg_facts):
+    """
+    @summary: Generate VNET and VXLAN config files and copy them to DUT.
+    @param duthost: DUT host object
+    @param mg_facts: Minigraph facts
+    """
+    logger.info("Generating config files and copying to DUT")
+
+    vnet_switch_config = [{
+        "SWITCH_TABLE:switch": {
+            "vxlan_port": vc.VXLAN_PORT,
+            "vxlan_router_mac": vc.VXLAN_MAC
+        },
+        "OP": "SET"
+    }]
+
+    duthost.copy(content=json.dumps(vnet_switch_config, indent=4), dest=vc.DUT_VNET_SWITCH_CONFIG)
+
+
+    render_template_to_host("vnet_vxlan.j2", duthost, vc.DUT_VNET_CONF, vc.VNET_CONFIG, mg_facts, ipv6_vxlan_test=vc.IPV6_VXLAN_TEST)
+    render_template_to_host("vnet_interface.j2", duthost, vc.DUT_VNET_INTF_CONFIG, vc.VNET_CONFIG)
+    render_template_to_host("vnet_nbr.j2", duthost, vc.DUT_VNET_NBR_JSON, vc.VNET_CONFIG)
+    render_template_to_host("vnet_routes.j2", duthost, vc.DUT_VNET_ROUTE_CONFIG, vc.VNET_CONFIG, op="SET")
+
+def apply_dut_config_files(duthost):
+    """
+    @summary: Applies config files on disk
+    @param duthost: DUT host object
+    """
+    logger.info("Applying config files on DUT")
+
+    config_files = ["/tmp/vnet.intf.json", "/tmp/vnet.nbr.json"]
+    if vc.APPLY_NEW_CONFIG:
+        config_files.append("/tmp/vnet.conf.json")
+
+    for config in config_files:
+        duthost.shell("sonic-cfggen -j {} --write-to-db".format(config))
+        sleep(3)
+
+    duthost.shell("docker cp {} swss:/vnet.route.json".format(vc.DUT_VNET_ROUTE_CONFIG))
+    duthost.shell("docker cp {} swss:/vnet.switch.json".format(vc.DUT_VNET_SWITCH_CONFIG))
+    if vc.APPLY_NEW_CONFIG:
+        duthost.shell("docker exec swss sh -c \"swssconfig /vnet.switch.json\"")
+        duthost.shell("docker exec swss sh -c \"swssconfig /vnet.route.json\"")
+        sleep(3)
+
+def cleanup_dut_vnets(duthost, mg_facts):
+    """
+    @summary: Removes all VNET information from DUT
+    @param duthost: DUT host object
+    @param mg_factS: Minigraph facts
+    """
+    logger.info("Removing VNET information from DUT")
+
+    for intf in vc.VNET_CONFIG['vlan_intf_list']:
+        duthost.shell("docker exec -i database redis-cli -n 4 del \"VLAN_INTERFACE|{}|{}\"".format(intf['ifname'], intf['ip']))
```

Comment:
Is there a reason we need to `docker exec` into the database or can we do this directly with `redis-cli` or `sonic-db-cli` (or whatever it's called these days)?

Reply:
@daall , as this is a conversion from ansible, followed the same as it was originally. Not sure we want to change this invocation unless it is widely changed all over

Reply:
`redis-cli` is used in the everflow tests. I'm OK with this as a direct conversion from ansible to pytest but in general it seems like the wrong way to handle this. @theasianpianist feel free to leave it for now but I would recommend coming back to it in a later PR. :)

Reply:
@prsunny would this test ever need to be run on a multi-ASIC platform? If so switching to `redis-cli` may be preferable since `redis-cli` should be re-written to account for multi-ASIC platforms, which might make adding support for this test easier in the future.

Reply:
it is not intended for multi-asic. but as Danny mentioned, since it redis-cli is used other places as well, we could change it to be consistent.

Reply:
Changed to use `redis-cli`